src/Field/Definition/ChoiceField.php

```diff
@@ -78,23 +78,29 @@ protected function toManagementApiData () : array
 
 	/**
 	 * @inheritDoc
+	 *
+	 * @param string|int|null|array<string|int|null> $data
 	 */
 	public function validateData (ComponentContext $context, array $contentPath, mixed $data, array $fullData) : void
 	{
 		if ($this->allowMultiselect)
 		{
+			\assert(null === $data || \is_array($data));
```

Comment:
Unfortunately this is not enough. This will produce an error 500 in prod, if something else comes in.

We really need to pass `mixed` through and validate it in the app.
Maybe split the validation in two steps:

1. base validation
2. normalization
3. remaining validation